tests/test_buffered_mode.py

```diff
@@ -165,6 +172,7 @@ def test_buffered_mode_change_buffer_size(self):
                 with signac.buffered(buffer_size=14):
                     pass
 
+    @pytest.mark.xfail(reason="This test uses various deprecated APIs.")
```

Comment:
Most of this test should still be viable. I'd rather see us rip out or alter the deprecated bits and keep the bulk of the code here, instead of `xfail`-ing it.

Reply:
While you're right that the test would still be viable if we modified it correctly, it would basically all be redundant. If you strip out the logic associated with the various deprecated buffer-related functions, then all of the behaviors are tested elsewhere. The only exception I see is the parts testing the amount of data stored in the buffer, and even those are only "new" in the sense that they'd be testing signac APIs that we would need to patch for the new classes, which already test the correct underlying behavior (i.e. there are synced collection tests verifying that buffers are flushed etc, so this test would just make sure that `signac.get_buffer_load() == JSONDict.get_buffer_size()` or so).

Reply:
But the purpose of an integration test is that you're testing all of it together. It doesn't matter that it's redundant, that's the point. 😆

Reply:
The thing you'd be integrating here is the test of buffering with different force modes. If `force_write` is no longer supported, there's nothing in this integration test that isn't tested (even in an integrated context) in other tests. Unless you think there's value in just checking this long sequence of changes?

Reply:
> Unless you think there's value in just checking this long sequence of changes?

That's all I'm going for. This test was added specifically to ensure that long sequences of changes work fine. I would just remove the `force_write` bits (probably just replacing them with standard non-forced calls, even if redundant).